packages/next/shared/lib/router/utils/prepare-destination.ts

```diff
@@ -191,14 +213,19 @@ export default function prepareDestination(
   }
 
   try {
-    newUrl = destinationCompiler(params)
+    newUrl = destPathCompiler(params)
 
     const [pathname, hash] = newUrl.split('#')
+    parsedDestination.hostname = destHostnameCompiler(params)
     parsedDestination.pathname = pathname
     parsedDestination.hash = `${hash ? '#' : ''}${hash || ''}`
     delete (parsedDestination as any).search
   } catch (err) {
-    if (err.message.match(/Expected .*? to not repeat, but got an array/)) {
+    if (
+      (err as Error).message.match(
```

Comment:
In the future we might want to use `if (err && err.message)` in case something in the try is throwing a non-error object such as `throw 'foo'`